### DISCUSSION TITLE
Problem with select_atoms

### DISCUSSION THREAD

**Discussion by katieesconfused** [MDAnalysis/mdanalysis#4587] [Q&A]
Dear MDAnalysis team and Users, 

I'm trying use MDAnalysis select_atoms function to dump out atom index, frame and positions of species at the Interface of a system. 
Problem : the output is WRONG, and very specifically wrong. 

This is my code: 
import MDAnalysis as mda
import numpy as np

topo   = "/scratch/anhnguyen99/g/s/si2/nvt/p10/ps.gro"
trjnj    = "/scratch/anhnguyen99/g/s/si2/nvt/p10/2nd10ns/ps2nj.xtc"
unj     = mda.Universe(topo, trjnj, format='xtc',topology_format="gro")
unj.trajectory[500]

linj     =  unj.select_atoms("name LI and (around 5 resname PE120)", periodic = False, updating= True)
print(linj.indices)

Result: 
[44338 44355 44405 44436 44446]

For demonstration I'm slicing at frame 500 in the trajectory.
The trajectory is converted to remove box periodicity using gromacs gmx trjconv -pbc nojump, 
This problem persist even when I use the original wrapped trajectory. 
 
When I visualize all of species that got selected this way, this is the result, which is INCORRECT: 
![Capture](https://github.com/MDAnalysis/mdanalysis/assets/151600565/4d30bb2f-a404-426f-b8b1-72ddd9ec0c32)

When I use _GROMACS gmx select_, I get these indices as results: 
44339 44406 44420 44441 44443 44445 44447

and this is what it looks like, which is CORRECT, for my purpose:
![corCapture](https://github.com/MDAnalysis/mdanalysis/assets/151600565/8b185de9-6911-4132-b58b-97d6e2196cf7)

The atoms that got selected _by MDAnalysis_ , for some of them, the index is off by exactly 1
44338 instead of  44339 
44405 instead of  44406 
44446 instead of  44447

I already set period =  False for geometric selection, and updating = True for dynamic selection in the trajectory. 
MDAnalysis version I'm using is MDAnalysis 2.7.0.

Any help is greatly appreciated!! If you have some time to spare please help and take a look at this, I suspect there is a bug with the function select_atoms of MDAnalysis 2.7.0. 

Thank you for your time, please let me know what other information I can provide to on this issue. 
Anh.

**Comment by hmacdope** [comment]
@katieesconfused 

A few issues, 
* you will need to set `periodic=True` when selecting with MDAnalysis otherwise some distances over the PBC will not be taken into account. 
* How are you visualizing? This is very important as could lead to a mismatch in the atoms selected.  Beware of any "off by one" bugs when comparing indices. MDAnalysis indicies are 0 indexed, (like python). I am not sure what the indexing is like in your visualization program.
